webhooks/webhooks.go

```diff
@@ -0,0 +1,227 @@
+package webhooks
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type Broadcaster interface {
+	BroadcastAction(ctx context.Context, action Action) error
+}
+
+const (
+	webhookTimeout   = 10 * time.Second
+	WebhookEventPing = "ping"
+)
+
+type (
+	Webhook struct {
+		Module string `json:"module"`
+		Event  string `json:"event"`
+		URL    string `json:"url"`
+	}
+
+	WebhookQueueInfo struct {
+		URL  string `json:"url"`
+		Size int    `json:"size"`
+	}
+
+	// Action describes an event that has been triggered.
+	Action struct {
+		Module  string      `json:"module"`
+		Event   string      `json:"event"`
+		Payload interface{} `json:"payload,omitempty"`
+	}
+)
+
+type Manager struct {
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+	logger    *zap.SugaredLogger
+	wg        sync.WaitGroup
+
+	mu       sync.Mutex
+	queues   map[string]*actionQueue // URL -> queue
+	webhooks map[string]Webhook
+}
+
+type actionQueue struct {
```

Comment:
`nit` I don't really feel strongly wrt `action` or `event` but this is another small indicator that we could perhaps squash both concepts or reverse them? An event queue sounds better than an action queue